lib/tool_shed/galaxy_install/tool_dependencies/recipe/step_handler.py

```diff
@@ -165,28 +166,13 @@ def url_download( self, install_dir, downloaded_file_name, download_url, extract
 
         # Set a timer so we don't sit here forever.
         start_time = time.time()
-        try:
-            src = urllib2.urlopen( download_url )
-            dst = open( file_path, 'wb' )
-            while True:
-                chunk = src.read( basic_util.CHUNK_SIZE )
-                if chunk:
-                    dst.write( chunk )
-                else:
-                    break
-                time_taken = time.time() - start_time
-                if time_taken > basic_util.NO_OUTPUT_TIMEOUT:
-                    err_msg = 'Downloading from URL %s took longer than the defined timeout period of %.1f seconds.' % \
-                        ( str( download_url ), basic_util.NO_OUTPUT_TIMEOUT )
-                    raise Exception( err_msg )
-        except Exception as e:
-            err_msg = err_msg = 'Error downloading from URL\n%s:\n%s' % ( str( download_url ), str( e ) )
-            raise Exception( err_msg )
-        finally:
-            if src:
-                src.close()
-            if dst:
-                dst.close()
+        src = requests.get(download_url, stream=True)
+        with open(file_path, 'wb') as dst:
+            for chunk in src.iter_content(chunk_size=basic_util.CHUNK_SIZE):
+                dst.write(chunk)
```

Comment:
This should be:
```python
if chunk:
    dst.write(chunk)
```
to filter out keep-alive new chunks.

Reply:
I copied this from @erasche's code in #4604 and still messed it up somehow... thanks as always for catching my mistakes.